xtask/src/tasks/fmt/house_rules/copyright.rs

```diff
@@ -26,37 +26,34 @@ pub fn check_copyright(path: &Path, fix: bool) -> anyhow::Result<()> {
     let first_line = lines.next().unwrap_or(Ok(String::new()))?;
     let second_line = lines.next().unwrap_or(Ok(String::new()))?;
     let third_line = lines.next().unwrap_or(Ok(String::new()))?;
-    let mut mit = false;
-
-    let missing_copyright = {
-        if first_line.contains(HEADER) {
-            // Correct Microsoft copyright.
-            false
-        } else if first_line.contains(HEADER_MIT_FIRST) && second_line.contains(HEADER_MIT_SECOND) {
-            // Some files may be Microsoft copyright with MIT license.
-            mit = true;
-            false
-        } else if first_line.contains("Copyright") && !first_line.contains("Microsoft") {
-            // OK, copyright someone else. Ignore this house rule.
-            return Ok(());
-        } else {
-            // Either the wrong Microsoft copyright, or no copyright.
-            true
-        }
-    };
 
-    let missing_blank_line = if mit {
-        !third_line.is_empty()
-    } else {
-        !second_line.is_empty()
-    };
+    // Preserve any files which are copyright, but not by Microsoft.
+    if first_line.contains("Copyright") && !first_line.contains("Microsoft") {
+        return Ok(());
+    }
+
+    let mut missing_banner =
+        !first_line.contains(HEADER_MIT_FIRST) || !second_line.contains(HEADER_MIT_SECOND);
+    let mut missing_blank_line = !third_line.is_empty();
+
+    // TEMP: until we have more robust infrastructure for distinct
+    // microsoft-internal checks, include this "escape hatch" for preserving
+    // non-MIT licensed files when running `xtask fmt` in the msft internal
+    // repo. This uses a job-specific env var, instead of being properly plumbed
+    // through via `clap`, to make it easier to remove in the future.
+    let is_msft_internal = std::env::var("XTASK_FMT_COPYRIGHT_ALLOW_MISSING_MIT").is_ok();
```

Comment:
Note that I've inculded this extra bit of logic here as well, so that `xtask fmt` continues to work in our closed-source code as well.

Lets get the repo compliant, and then we can work on making this a bit more elegant in a follow-up.